weighted_levenshtein/clev.pyx

```diff
@@ -179,24 +188,33 @@ def damerau_levenshtein(
     if transpose_costs is None:
         transpose_costs = unit_matrix
 
-    s1 = str(str1).encode()  
-    s2 = str(str2).encode()  
+    s1 = str1.encode('utf-8').decode('utf-8')
```

Comment:
Could we maybe add a comment explaining what this part is doing? There are a lot of subtleties in Cython string implicit conversion (and frankly I can't really tell what this is doing either without digging through a lot of documentation).

Also it would be nice to create a helper function for string conversion that can be shared between lev, dam_leve, and osa. Maybe we can document what this does in that helper function